### PR TITLE
Updated AzAttestation cmdlets for public preview

### PR DESCRIPTION
**Pull request by gkostal**
## Description

Update ```AzAttestation``` cmdlets for public preview.
- Added policy signer management cmdlets
- Added support for location and tags

The cmdlet changes made by this PR have been approved here:
- [Add Trusted Policy Signers (Issue #484)](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/484)
- [Add Location and Tags to Attestation Provider (Issue #491)](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/491)

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Attestation/Attestation/Commands/AddAzAttestationPolicySigner.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+using Microsoft.Azure.Commands.Attestation.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Attestation
+{
+    /// <summary>
+    /// Set AttestationPolicy.
+    /// </summary>
+    [Cmdlet("Add", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AttestationPolicySigner", SupportsShouldProcess = true)]
+    [OutputType(typeof(String))]
```
Comment (VeryEarly) [inline src/Attestation/Attestation/Commands/AddAzAttestationPolicySigner.cs]: Based on [cmdlet design guide line](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/cmdlet-best-practices.md), the output type should never be type of String.

We encourage to have a wrapper class to encapsulate the string message

File: src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Attestation.Models;
+
+namespace Microsoft.Azure.Commands.Attestation
+{
+    /// <summary>
+    /// Get AttestationPolicy.
+    /// </summary>
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AttestationPolicySigners", SupportsShouldProcess = true)]
```
Comment (VeryEarly) [inline src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs]: no need to have SupportsShouldProcess for "Get-*"

File: src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    [OutputType(typeof(String))]
```
Comment (VeryEarly) [inline src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs]: output type should be PSPolicySigners

File: src/Attestation/Attestation/Commands/RemoveAzAttestationPolicySigner.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+using Microsoft.Azure.Commands.Attestation.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Attestation
+{
+    /// <summary>
+    /// Set AttestationPolicy.
+    /// </summary>
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AttestationPolicySigner", SupportsShouldProcess = true)]
+    [OutputType(typeof(String))]
```
Comment (VeryEarly) [inline src/Attestation/Attestation/Commands/RemoveAzAttestationPolicySigner.cs]: output type should be PSPolicySigners

File: src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            HelpMessage = "Specifies the ResourceID of an attestation provider.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            String policySignersString = AttestationDataPlaneClient.GetPolicySigners(Name, ResourceGroupName, ResourceId);
+            PSPolicySigners policySigners = new PSPolicySigners(policySignersString);
```
Comment (VeryEarly) [inline src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs]: Is it possible the "policySignersString" contains multiple signers and need to be placed into array of PSPolicySigners ?

File: src/Attestation/Attestation/help/Add-AzAttestationPolicySigner.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+### CommonParameters
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
+
+## INPUTS
+
+### System.String
+
+## OUTPUTS
+
+### System.String
```
Comment (VeryEarly) [inline src/Attestation/Attestation/help/Add-AzAttestationPolicySigner.md]: update output type

File: src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+Aliases:
+
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -Confirm
```
Comment (VeryEarly) [inline src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md]: this comes from SupportShouldProcess I believe

File: src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+```
+
+### CommonParameters
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
+
+## INPUTS
+
+### System.String
+
+## OUTPUTS
```
Comment (VeryEarly) [inline src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md]: update the output type

File: src/Attestation/Attestation/help/Remove-AzAttestationPolicySigner.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+### CommonParameters
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
+
+## INPUTS
+
+### System.String
+
+## OUTPUTS
```
Comment (VeryEarly) [inline src/Attestation/Attestation/help/Remove-AzAttestationPolicySigner.md]: update the output type

File: src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs (hunk 5 above)
Comment (gkostal) [reply src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs]: The PSPolicySigners class represents a collection of signers via the Certificates property:
```
public string[] Certificates { get; protected set; }
```
The Certificates array will contain the correct number of certificates (aka signers) contained in the "policySignersString" string.

File: src/Attestation/Attestation/Commands/AddAzAttestationPolicySigner.cs (hunk 1 above)
Comment (gkostal) [reply src/Attestation/Attestation/Commands/AddAzAttestationPolicySigner.cs]: Thanks, fixed.

File: src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs (hunk 2 above)
Comment (gkostal) [reply src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs]: Thanks, fixed.

File: src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs (hunk 3 above)
Comment (gkostal) [reply src/Attestation/Attestation/Commands/GetAzureAttestationPolicySigners.cs]: Thanks, fixed.

File: src/Attestation/Attestation/Commands/RemoveAzAttestationPolicySigner.cs (hunk 4 above)
Comment (gkostal) [reply src/Attestation/Attestation/Commands/RemoveAzAttestationPolicySigner.cs]: Thanks, fixed.

File: src/Attestation/Attestation/help/Add-AzAttestationPolicySigner.md (hunk 6 above)
Comment (gkostal) [reply src/Attestation/Attestation/help/Add-AzAttestationPolicySigner.md]: Thanks, fixed.

File: src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md (hunk 7 above)
Comment (gkostal) [reply src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md]: Thanks, fixed.

File: src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md (hunk 8 above)
Comment (gkostal) [reply src/Attestation/Attestation/help/Get-AzAttestationPolicySigners.md]: Thanks, fixed.

File: src/Attestation/Attestation/help/Remove-AzAttestationPolicySigner.md (hunk 9 above)
Comment (gkostal) [reply src/Attestation/Attestation/help/Remove-AzAttestationPolicySigner.md]: Thanks, fixed.

**Pull request merged**